### PR TITLE
add conflict detection to assignment scripts

### PR DESCRIPTION
**Pull request by mspector**
for each of "assign-program-committee" and "assign-senior-program-committee", the script now finds all the conflict domains for the authors of the paper, and adds those domains as nonreaders to the groups Paper#/Reviewers, Paper#/Area_Chair and Paper#/AnonReviewer#

it also alerts you if you try to assign a reviewer or area chair to a paper with which they have a conflict, and gives you the option to either remove the conflict and continue, or to cancel the assignment (and move on, if you're assigning a list of users). This caused many problems for ICLR, because the user was not a reader of their own Reviewer or Area Chair group.

### CODE REVIEW

File: uai2017/python/assign-program-committee.py [hunk 1]
```diff
@@ -49,7 +49,25 @@ def single_assignment_valid(s):
     except IndexError:
         return False
 
-def assign_reviewer(reviewer,paper_number):
+def get_nonreaders(paper_number):
+    # nonreaders are:
+    # (1) all authors of the paper
+    # (2) all domain groups of the authors of the paper
```
Comment (melisabok) [inline uai2017/python/assign-program-committee.py]: I think adding only the domains is enough, right?

File: uai2017/python/assign-program-committee.py [hunk 2]
```diff
@@ -137,22 +159,42 @@ def clear_assignments():
...
-        print "Invalid input"
-        sys.exit()
+
+def assign(assignments):
+    for a in assignments:
+        reviewer = a[0]
+        paper_number = a[1]
+        conflict_list = get_nonreaders(paper_number)
+
+        members = set([g.id for g in client.get_groups(member=reviewer)])
```
Comment (melisabok) [inline uai2017/python/assign-program-committee.py]: I like this!

File: uai2017/python/assign-senior-program-committee.py [hunk 3]
```diff
@@ -49,7 +49,25 @@ def single_assignment_valid(s):
     except IndexError:
         return False
 
-def assign_areachair(areachair, paper_number):
+def get_nonreaders(paper_number):
```
Comment (melisabok) [inline uai2017/python/assign-senior-program-committee.py]: we need to start to think about the duplicated code.

File: uai2017/python/assign-program-committee.py (hunk 1 above)
Comment (mspector) [inline uai2017/python/assign-program-committee.py]: oops, the comment is out of date. it does only do domains

File: uai2017/python/assign-senior-program-committee.py (hunk 3 above)
Comment (mspector) [inline uai2017/python/assign-senior-program-committee.py]: yes, agreed :/

**Pull request merged**